### ISSUE TITLE
Weird spacing effects with ardhavisarga in Noto Serif Devanagari

### ISSUE THREAD

**Issue by NorbertLindenberg** [googlefonts/noto-fonts#1822]
Noto Serif Devanagari (version 2.001) shows the characters U+1CF2 VEDIC SIGN ARDHAVISARGA and U+1CF3 VEDIC SIGN ROTATED ARDHAVISARGA with different weird spacing effects in different browsers.

To reproduce, go to
https://lindenbergsoftware.com/google/noto/deva-ardhavisarga.html

The columns for Sans and Serif contain the same character sequences, five times each the characters mentioned above, first by themselves, then bracket by some other Devanagari characters. In the Sans column (using Noto Sans Devanagari), the character are shown five times. In the Serif column (using Noto Serif Devanagari), it often appears that there’s only one character.

Microsoft Edge 44.19041.1.0; EdgeHTML 18.19041; Microsoft Windows 10 version 2004:

<img width="434" alt="Screen Shot 2020-08-19 at 23 10 45" src="https://user-images.githubusercontent.com/216206/90723254-44b33b00-e271-11ea-82c1-819097f0ea6a.png">

Safari 13.1.2 on macOS 10.15.6:

<img width="385" alt="Screen Shot 2020-08-19 at 23 06 11" src="https://user-images.githubusercontent.com/216206/90723485-aa9fc280-e271-11ea-9028-e0d77b2b00d6.png">

Firefox 79.0 on macOS 10.14.6:

<img width="299" alt="Screen Shot 2020-08-19 at 23 13 35" src="https://user-images.githubusercontent.com/216206/90723460-a2478780-e271-11ea-9240-832860dcfcfe.png">

**Comment by nizarsq** [comment]
@NorbertLindenberg could you try to reproduce this issue using fonts in the following directories: 
https://github.com/googlefonts/noto-fonts/tree/master/phaseIII_only/unhinted/ttf/NotoSansDevanagari/NotoSansDevanagari-Regular.ttf
https://github.com/googlefonts/noto-fonts/tree/master/phaseIII_only/unhinted/ttf/NotoSansDevanagariUI/NotoSansDevanagariUI-Regular.ttf
https://github.com/googlefonts/noto-fonts/tree/master/phaseIII_only/unhinted/ttf/NotoSerifDevanagari/NotoSerifDevanagari-Regular.ttf

I couldn't reproduce it on Firefox, Chrome  or Safari.
<img width="1034" alt="Screen Shot 2020-08-20 at 5 08 57 PM" src="https://user-images.githubusercontent.com/22158882/90837768-d79eb480-e307-11ea-834c-b9c3fdfb6a48.png">

**Comment by NorbertLindenberg** [comment]
Those are the exact versions that I'm using, except I'm not using the UI flavor.

When you point your browsers to my web page, do you see what I'm seeing? Can you try just the sequences of ardhavisarga in your test environment, without क and danda?

**Comment by nizarsq** [comment]
Yes I see what you see using https://lindenbergsoftware.com/google/noto/deva-ardhavisarga.html:
<img width="1270" alt="Screen Shot 2020-08-20 at 5 44 54 PM" src="https://user-images.githubusercontent.com/22158882/90839638-dcb23280-e30c-11ea-81ca-6676d3d5d5d4.png">




 I think the issue has to do with your code structure. I modified your code and it seems to work just fine:
<img width="1271" alt="Screen Shot 2020-08-20 at 5 40 43 PM" src="https://user-images.githubusercontent.com/22158882/90839390-4847d000-e30c-11ea-9d07-750f2d88cb91.png">

**Comment by nizarsq** [comment]
I used blank fallback font just to make sure browser will not use another fallback font,

**Comment by punchcutter** [comment]
Both of those are classified as marks in the GDEF so width will get zeroed in some places and not others. They should be changed to GDEF class 1 in the source. I just did it and rebuilt and it looks fine.

**Comment by NorbertLindenberg** [comment]
@nizarsq The modified HTML code nests `<th>` inside `<span>`, which isn't valid, and I wouldn't say that it works fine with all the tofu in the rendering.

@punchcutter This sounds reasonable to me.

**Comment by nizarsq** [comment]
> @nizarsq The modified HTML code nests `<th>` inside `<span>`, which isn't valid, and I wouldn't say that it works fine with all the tofu in the rendering.
> 
> @punchcutter This sounds reasonable to me. Well, other than the mark-width-zeroing business being entirely unreasonable...

@NorbertLindenberg Sorry I attached a wrong screenshot. It uses the browser font and that. explain the tufo.  The is the test case I created
<img width="1433" alt="Screen Shot 2020-08-20 at 7 08 05 PM" src="https://user-images.githubusercontent.com/22158882/90844319-7d5a1f80-e318-11ea-8bb6-52641b29b145.png">

**Comment by punchcutter** [comment]
There are actually a few glyphs in here with the wrong GDEF category. And they don't match the GDEF of Sans Devanagari. I can send a PR with fixes.
